### ISSUE TITLE
[Bug]: OpenCV issues with cmake and pkg-config

### ISSUE THREAD

**Issue by jonas-ott** [termux/termux-packages#10886]
### Problem description

In `/data/data/com.termux/files/usr/lib/pkgconfig/` should be an `opencv4.pc` file. Without it `pkg-config --cflags --libs opencv4` returns `No package 'opencv4' found'. 

Cmake also has problems finding OpenCv:
```
CMake Warning at /data/data/com.termux/files/usr/lib/cmake/opencv4/OpenCVConfig.cmake:116 (message):
OpenCV: Include directory doesn't exist: '//include/opencv4'.  OpenCV
installation may be broken.  Skip...
Call Stack (most recent call first):
CMakeLists.txt:3 (find_package)
```
If you change Line 116 in `OpenCVConfig.cmake` from
 `set(\__OpenCV_INCLUDE_DIRS "${OpenCV_INSTALL_PATH}/include/opencv4")` to 
`set(\__OpenCV_INCLUDE_DIRS "/data/data/com.termux/files/usr/include/opencv4")` everything works fine. 

 

### What steps will reproduce the bug?

1. `pkg install opencv`

for pkg-config:
1.  pkg-config --cflags --libs opencv4`

for CMake:
1. create a `CMakeLists.txt` file with following content 
```
cmake_minimum_required(VERSION 2.8)
project( DisplayImage )
find_package( OpenCV REQUIRED )
include_directories( ${OpenCV_INCLUDE_DIRS} )
add_executable( DisplayImage DisplayImage.cpp )
target_link_libraries( DisplayImage ${OpenCV_LIBS} )
```
2. `cmake ./`

### What is the expected behavior?

pkg-config finds OpenCV and returns the expected flags.
CMake runs without an error, as with the modified `OpenCVConfig.cmake` file

### System information

termux-info:

```
Termux Variables:
TERMUX_API_VERSION=0.50.1
TERMUX_APK_RELEASE=F_DROID
TERMUX_APP_PACKAGE_MANAGER=apt
TERMUX_APP_PID=14664
TERMUX_IS_DEBUGGABLE_BUILD=0
TERMUX_MAIN_PACKAGE_FORMAT=debian
TERMUX_VERSION=0.118.0
Packages CPU architecture:
aarch64
Subscribed repositories:
\# sources.list
deb <https://packages-cf.termux.org/apt/termux-main> stable main
Updatable packages:
All packages up to date
termux-tools version:
0\.184
Android version:
12
Kernel build information:
Linux [localhost](http://localhost) 4.14.113-23851207 #1 SMP PREEMPT Wed Mar 9 18:04:34 KST 2022 aarch64 Android
Device manufacturer:
samsung
Device model:
SM-P610
```

**Comment by RalfWerner** [comment]
I've the same problem and device (`SM-P610`) and [here](https://github.com/termux/termux-packages/pull/8392#issuecomment-1146232419) described an example. The error indicates _missing_ `GTK` in [build](https://github.com/termux/termux-packages/blob/master/packages/opencv/build.sh#L12).

**Comment by RalfWerner** [comment]
@Grimler91 why off-topic?

**Comment by Grimler91** [comment]
Your permission issue in the other thread (which by the way is a closed pull request, please open an actual issue and fill in the bug report template if there is a problem) doesn't look related to the missing cmake/pkg-config script reported here

**Comment by Grimler91** [comment]
> But then continued the test without a `pkg-config` and get the `GTK` error

The error at the end says that opencv was built without qt or gtk support. It has nothing in common with the issue here (which is that an opencv pkg-config script is not installed by our build recipe)

> On the other hand, if it can be solved, the camera preview would then be an _OpenCV_ problem again. If the simple `*.jpg` application is solved, I would open an issue in **termux-api** if this has a _**chance of success!?**_

I have never used opencv, so I have no idea what capabilities it needs to function. You seem familiar with it and probably have a better idea about the requirements than me, so please open an issue in termux-api, fill in the bug report template, and explain what opencv needs for it to work. If it just needs a .jpg from the camera (rather than access to the underlying camera hardware) then it might be possible to make it work without too much work, since we have termux-camera-photo for that.

**Comment by RalfWerner** [comment]
> I have **never used** opencv ... we have **termux-camera-photo** `(*.jpg)`  for that.

I also **not!** ... **That** I've used in [Example](https://github.com/termux/termux-packages/pull/8392#issuecomment-1146460627). You say: _opencv was built without qt or **gtk** support_ and _**pkg-config** script is not installed._
`opencv` without `GTK` makes no sence. So we're again in **this** issue and my question was: _**What is to do to fix that?**_

If you have no experience with `opencv`, possibly the authors: @MrAdityaAlok or @xtkoba may answer.

**Comment by Grimler91** [comment]
> opencv*) without GTK makes no sence

Why not? Looking at your example code again it looks like the thing that fails is to create a window (`namedWindow(A[k])`), so seems to me that the package might work as long as GUI elements are avoided

> What is to do to fix that?

Adding relevant qt or gtk packages to dependencies of opencv, and fixing potential build errors that appear

**Comment by Grimler91** [comment]
> Looking at your example code again it looks like the thing that fails is to create a window (namedWindow(A[k])), so seems to me that the package might work as long as GUI elements are avoided

Reading up on opencv shows that this is the case. HighGUI (i.e. the "high level GUI") is the only part of opencv that depends on qt5/gtk, so opencv should work fine if the HighGUI module is avoided (meaning then also that you can't create GUI elements)

**Comment by Grimler91** [comment]
Should be fixed in 4.5.5-2, thanks for reporting

**Comment by jonas-ott** [comment]
Thank you for the quick fix, now pkg-config works as expected.
